protocol/metrics/consumer_relayserver_client.go

```diff
@@ -0,0 +1,196 @@
+package metrics
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/lavanet/lava/utils"
+)
+
+type ConsumerRelayserverClient struct {
+	endPointAddress    string
+	addQueue           []UpdateMetricsRequest
+	sendQueue          []UpdateMetricsRequest
+	ticker             *time.Ticker
+	lock               sync.Mutex
+	sendID             int
+	isSendQueueRunning bool
+}
+type UpdateMetricsRequest struct {
+	RecordDate      string `json:"RecordDate"`
+	Hash            string `json:"Hash"`
+	Chain           string `json:"Chain"`
+	ApiType         string `json:"ApiType"`
+	RelaysInc       uint64 `json:"RelaysInc"`
+	CuInc           int    `json:"CuInc"`
+	LatencyToAdd    uint64 `json:"LatencyToAdd"`
+	LatencyAvgCount int    `json:"LatencyAvgCount"`
+}
+
+func NewConsumerRelayserverClient(endPointAddress string) *ConsumerRelayserverClient {
+	if endPointAddress == DisabledFlagOption {
+		utils.LavaFormatDebug("CUC: Consumer Usageserver disabled")
+		return nil
+	}
+
+	cuc := &ConsumerRelayserverClient{
+		endPointAddress: endPointAddress,
+		ticker:          time.NewTicker(30 * time.Second),
+		sendQueue:       make([]UpdateMetricsRequest, 0),
+		addQueue:        make([]UpdateMetricsRequest, 0),
+		lock:            sync.Mutex{},
+	}
+
+	utils.LavaFormatDebug("CUC: Starting processQueue goroutine")
+	go cuc.processQueue()
+
+	return cuc
+}
+
+func (cuc *ConsumerRelayserverClient) processQueue() {
+	if cuc == nil {
+		return
+	}
+
+	for range cuc.ticker.C {
+		cuc.lock.Lock()
```

Comment:
thats a really bad usage of locks. 

best practice of locks is to have wrapper methods that lock and defer unlock. 
do what you need in the method and return. 
there shouldn't be multiple locking and unlocking in the same function body that's a garenteed deadlock recepie